File: FinanceApp/FinanceApp/Models/Transaction.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using FinanceApp.Enums;
using System.Text.Json.Serialization;

namespace FinanceApp.Models
{
    public class Transaction
    {
        [Required] [Key] public int Id { get; set; }

        [Required] public string Notice { get; set; }

        [Required] public TransactionType Type { get; set; }

        [Required] public int Price { get; set; }

        [Required] public DateTime DateOfTransaction { get; set; }

        [DefaultValue(false)] public bool IsCategorized { get; set; }

        public virtual Category? Category { get; set; }
        [ForeignKey("Category")] public int? CategoryId { get; set; }

        [JsonIgnore] public Wallet? Wallet { get; set; }
        [ForeignKey("Wallet")] public int WalletId { get; set; }
    }
}

File: FinanceApp/FinanceApp/ClientApp/src/app/models/Transaction.ts
import {Category} from "./Category";

export class Transaction {
  constructor(id: string, type: TransactionType, notice: string, price: number, category: Category, dateOfTransaction: string) {
    this.id = id;
    this.type = type;
    this.notice = notice;
    this.price = price;
    this.category = category;
    this.dateOfTransaction = dateOfTransaction;
  }

  id: string;
  type: TransactionType;
  notice: string;
  price: number;
  category: Category;
  dateOfTransaction: string;
}

export enum TransactionType {
  Expense = "EXPENSE",
  Income = "INCOME"
}
